common/namer/namer_test.go

```diff
@@ -0,0 +1,173 @@
+package namer
+
+import (
+	"math"
+	"strconv"
+	"strings"
+	"testing"
+
+	fuzz "github.com/google/gofuzz"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestJoinWithMaxLength(t *testing.T) {
+	for i, tt := range []struct {
+		maxLength int
+		tokens    []string
+		noHash    bool
+		expected  string
+	}{
+		// No truncation needed
+		{
+			maxLength: math.MaxInt,
+			tokens:    []string{"foo", "bar", "baz"},
+			expected:  "foo-bar-baz",
+		},
+		// Max length too small
+		// Defaults to hash only
+		{
+			maxLength: 4,
+			tokens:    []string{"foo", "bar", "baz", "qux", "quux"},
+			expected:  "1008",
+		},
+		// Truncation is applied to tokens proportionally to their size
+		{
+			maxLength: 23,
+			tokens:    []string{"FfffOoooOooo", "BbbAaaRrr", "BbAaZz", "QUX"},
+			noHash:    true,
+			expected:  "FfffOooo-BbbAaa-BbAa-QU",
+		},
+		{
+			maxLength: 13,
+			tokens:    []string{"FfffOoooOooo", "BbbAaaRrr", "BbAaZz", "QUX"},
+			noHash:    true,
+			expected:  "Ffff-Bbb-Bb-Q",
+		},
+		// Truncation are spread evenly on all tokens
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
```suggestion
		// Truncation are spread best effort evenly on all tokens
```